shells.nix

```diff
@@ -32,57 +32,49 @@ in
           mkShellForComponent = component: attributePath:
             builtins.mapAttrs
               (name: drv:
-                let
-                  # add shell inputs and commands to native build inputs
-                  # note that this has to run after the shell derivation is created to let
-                  # it first run its logic on nativeBuildInputs ++
-                  # checkInputs/installCheckInputs
-                  #
-                  # Priority of nativeBuildInputs then becomes:
-                  #  1. nativeBuildInputs
-                  #  2. checkInputs/installCheckInputs
-                  #  3. shellInputs
-                  #  4. shellCommands
-                  addShellInputs = drv: drv.overrideAttrs (oldAttrs:
-                    let
-                      shellCommandAttrsOrDrv = oldAttrs.passthru.shellCommands or
-                        oldAttrs.shellCommands or { };
-                      shellCommands =
-                        if lib.isDerivation shellCommandAttrsOrDrv then
-                          shellCommandAttrsOrDrv
-                        else
-                          mkShellCommands oldAttrs.name shellCommandAttrsOrDrv;
-                    in
-                    {
-                      inherit shellCommands;
-                      nativeBuildInputs = oldAttrs.nativeBuildInputs or [ ]
-                      ++ oldAttrs.passthru.shellInputs or [ ]
-                      ++ oldAttrs.shellInputs or [ ]
-                      ++ [ shellCommands ];
-                    });
+                drv.overrideAttrs (oldAttrs:
+                  let
+                    targetName = lib.escapeShellArg ''target "\x1b[1m${name}\x1b[0m" in "\x1b[1m${builtins.concatStringsSep "." attributePath}\x1b[0m"'';
+                    shellCommandAttrsOrDrv = oldAttrs.passthru.shellCommands or
+                      oldAttrs.shellCommands or { };
+                    shellCommands =
+                      if lib.isDerivation shellCommandAttrsOrDrv then
+                        shellCommandAttrsOrDrv
+                      else
+                        mkShellCommands oldAttrs.name shellCommandAttrsOrDrv;
+                  in
+                  {
+                    inherit shellCommands;
+                    name = "${component.name}-${oldAttrs.name}-shell";
 
-                  targetName = lib.escapeShellArg ''target "\x1b[1m${name}\x1b[0m" in "\x1b[1m${builtins.concatStringsSep "." attributePath}\x1b[0m"'';
-                  shellPkg = (drv.drvAttrs // {
-                    inherit (drv) passthru;
-                    name = "${component.name}-${drv.name}-shell";
+                    # add shell inputs and commands to native build inputs
+                    # note that this has to run after the shell derivation is created to let
```

Comment:
This line isn't really true any more? Without mkShell there's no shell derivation...